app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -1055,4 +1060,41 @@ private function dispatchCartDataEvent($eventName, $quote, &$cartSubmissionData)
         );
         $cartSubmissionData = $cartDataWrapper->getCartData();
     }
+
+    /**
+     * It is Magento part of code which allow to work
+     * with applied Catalog Price Rule in the
+     * back-office due order create process.
+     * The original path is \Mage_Adminhtml_Sales_Order_CreateController::_processActionData()
+     * and \Mage_Adminhtml_Model_Sales_Order_Create::initRuleData()
+     *
+     * @param $storeId
+     * @param $websiteId
+     * @param $customerGroupId
+     */
+    protected function initRuleData($storeId, $websiteId, $customerGroupId)
+    {
+        Mage::register('rule_data', new Varien_Object(array(
+            'store_id' => $storeId,
+            'website_id' => $websiteId,
+            'customer_group_id' => $customerGroupId,
+        )));
+    }
+
+    /**
+     * @return bool
+     */
+    private function isAdmin()
+    {
+        return (bool) Mage::app()->getStore()->isAdmin();
+    }
```

Comment:
We are defining this function way too much and inconsistently.  Another definition for it is here:
https://github.com/BoltApp/bolt-magento1/blob/master/app/code/community/Bolt/Boltpay/Model/Payment.php#L144-L150

Can we refactor both to a singular implementation and place it here:
https://github.com/BoltApp/bolt-magento1/blob/master/app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php#L148'

By placing this int the General Trait, we will in turn make this general method globally available to all of our classes.  The current `Bolt_Boltpay_Model_Payment::isAdminArea` can in turn be a wrapper to the GeneralTrait method.